### ISSUE TITLE
Is MolGAN incompatible with max_atoms=31 or ZINC250k dataset

### ISSUE THREAD

**Issue by UmarZein** [deepchem/deepchem#3962]
## ❓ Questions & Help

I have replicated 90% of https://deepchem.io/tutorials/generating-molecules-with-molgan/

I assume it's more of a problem on the model/architecture rather than the library's implementation, but if anyone has used this library's MolGAN to fit an at least equally complex dataset please say so

with the differences being:

- i am using ZINC250k https://raw.githubusercontent.com/aspuru-guzik-group/chemical_vae/master/models/zinc_properties/250k_rndm_zinc_drugs_clean_3.csv
- `num_atoms=31` (95% of the molecules in the dataset has at most 31 atoms)
- `atom_labels=[0, 6, 7, 8, 9, 15, 16, 17, 35, 53]`
- tried `MolGAN(learning_rate=ExponentialDecay(0.0001, 0.9, 5000), vertices=max_atoms, model_dir="model_dir")`
- tried `MolGAN(learning_rate=ExponentialDecay(0.001, 0.9, 5000), vertices=max_atoms, model_dir="model_dir")`
- code to fit `gan.fit_gan(iterbatches(25), generator_steps=0.2, checkpoint_interval=1000, max_checkpoints_to_keep=25, restore=False)`
- `dataset.X.shape` is `(235043, 31, 31)`
- `dataset.y.shape` is `(235043, 31)`

**Comment by rbharath** [comment]
CC @shreyasvinaya who may have some insights

**Comment by shreyasvinaya** [comment]
Hi @UmarZein 
The Losses reported on the terminal are actually the sum and not the loss of the step, it is relatively harmless, please run it completely and let me know if it runs successfully

**Comment by UmarZein** [comment]
I haven't been able to get a good run
Previously, it experienced mode collapse, but today i tried it again and it just failed generation completely
notebook: https://gist.github.com/UmarZein/d9613e44922b2cba4199f8175d3b89f2
_I am not familiar with tensorflow_: I am using runpod.io's Tensorflow image/template and there were a couple of red texts there:
when i `import tensorflow as tf; print(tf.config.list_physical_devices('GPU'))`
```error
2024-04-30 06:38:19.185759: E tensorflow/compiler/xla/stream_executor/cuda/cuda_dnn.cc:9342] Unable to register cuDNN factory: Attempting to register factory for plugin cuDNN when one has already been registered
2024-04-30 06:38:19.185804: E tensorflow/compiler/xla/stream_executor/cuda/cuda_fft.cc:609] Unable to register cuFFT factory: Attempting to register factory for plugin cuFFT when one has already been registered
2024-04-30 06:38:19.186730: E tensorflow/compiler/xla/stream_executor/cuda/cuda_blas.cc:1518] Unable to register cuBLAS factory: Attempting to register factory for plugin cuBLAS when one has already been registered
2024-04-30 06:38:19.291660: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
```
when running `gan = MolGAN(learning_rate=ExponentialDecay(0.0001, 0.9, 5000), vertices=max_atoms, model_dir="model_dir")`
```
2024-04-30 06:53:18.816166: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1886] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 18452 MB memory:  -> device: 0, name: NVIDIA RTX 4000 Ada Generation, pci bus id: 0000:82:00.0, compute capability: 8.9
```
^after that line, it eats 18GB of VRAM (dunno if this is it's actual size or whether there is something wrong with the docker container)

**Comment by rbharath** [comment]
@shreyasvinaya Is the new torch molgan stable to run?

**Comment by shreyasvinaya** [comment]
@UmarZein The torch version of MolGAN is more stable to run compared to the TF version, I think you might have hit a state of model collapse, i would suggest you to try running it a couple of times, atleast one of them would give some good results
